training/security.md

```diff
@@ -0,0 +1,166 @@
+---
+title: Secure Your Cluster
+summary: A secure CockroachDB cluster uses TLS for encrypted inter-node and client-node communication and requires CA, node, and client certificates and keys.
+toc: false
+sidebar_data: sidebar-data-training.json
+---
+
+<div id="toc"></div>
+
+## Presentation
+
+<iframe src="https://docs.google.com/presentation/d/e/2PACX-1vSozBu9RIF-1V73oPtK-HTJpdtG3euC4oS8aR2Ze0AmIkBcTr1pKjblJ9Q6TJBLbREEX8wmcRkzrLeq/embed?start=false&loop=false" frameborder="0" width="756" height="454" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>
+
+#### URL for comments
+
+[Secure Your Cluster](https://docs.google.com/presentation/d/1RqiRikOPMHdtynASNdl20Vfxxd764vuNFRaxCVeXLB4/)
+
+## Lab
+
+Up to this point, we've been using an insecure local cluster. However, when using CockroachDB on remote servers, it's necessary to secure your deployment––when CockroachDB runs in insecure mode, it's _really_ insecure.
+
+In this lab, we'll show you how to secure your cluster.
+
+### Before You Begin
+
+To complete this lab, you need to have the [`cockroach` binary installed](/stable/install-cockroachdb.html).
+
+We also assume you have a local, insecure 3-node cluster running. If you don't, you can skip the **Tear down existing cluster** section of this lab.
+
+### Overview
+
+1. Tear down existing cluster.
+2. Generate certificate files.
+3. Restart cluster with SSL encryption.
+4. Connect to the cluster using encryption.
+
+### Step 1. Tear Down Cluster
+
+You can do this elegantly by using `cockroach quit` on each node, but we're going to close down everything in one fell swoop:
+
+{% include copy-clipboard.html %}
+~~~ shell
+$ pkill -9 cockroach
```

Comment:
I *guess* this one is ok.

Reply:
Actually, no, it'd be better to go for a clean shutdown here. The reason is that we're really *upgrading* this cluster to be secure, and in this scenario you'd prefer cleanly shutting down the cluster. The complication is that some of the nodes won't shut down cleanly (loss of quorum, etc), but then a double `pkill cockroach` would still be better and also serves as an opportunity to say why the second one is "needed" (just to save time waiting for the hard shutdown).

Reply:
Going for 2 `pkill`s for the sake of convenience. Adding explanatory text, as well.